flutter_local_notifications/lib/src/platform_flutter_local_notifications.dart

```diff
@@ -660,3 +663,164 @@ class MacOSFlutterLocalNotificationsPlugin
+    ? <String, LinuxNotificationButtonHandler> {
+      for (final LinuxNotificationButton button in notificationDetails.buttons)
+        button.buttonId : button.handler,
+    } : null;
+    return _channel.invokeMethod('show', <String, Object> {
+      'id': id,
+      'title': title,
+      'body': body,
+      'payload': payload ?? '',
+      'platformSpecifics': notificationDetails?.toMap(),
+    });
+  }
+
+  @override
+  Future<void> periodicallyShow(
+    int id,
+    String title,
+    String body,
+    RepeatInterval repeatInterval, {
+    LinuxNotificationDetails notificationDetails,
+    String payload,
+  }) {
+    validateId(id);
+    _periodicNotificationButtonHandlerMap[id] = notificationDetails != null
+    ? <String, LinuxNotificationButtonHandler> {
+      for (final LinuxNotificationButton button in notificationDetails.buttons)
+        button.buttonId : button.handler,
+    } : null;
+    return _channel.invokeMethod('show', <String, Object> {
+      'id': id,
+      'title': title,
+      'body': body,
+      'repeatInterval': repeatInterval.index,
+      'payload': payload ?? '',
+      'platformSpecifics': notificationDetails?.toMap(),
+    });
+  }
+
+  /// Schedules a notification to be shown at the specified date and time
+  /// relative to a specific time zone.
+  Future<void> zonedSchedule(
+    int id,
+    String title,
+    String body,
+    TZDateTime scheduledDate,
+    LinuxNotificationDetails notificationDetails, {
+    String payload,
+    DateTimeComponents matchDateTimeComponents,
+  }) async {
+    validateId(id);
+    validateDateIsInTheFuture(scheduledDate);
+    _periodicNotificationButtonHandlerMap[id] = notificationDetails != null
+    ? <String, LinuxNotificationButtonHandler> {
+      for (final LinuxNotificationButton button in notificationDetails.buttons)
+        button.buttonId : button.handler,
+    } : null;
+    final Map<String, Object> serializedPlatformSpecifics
+      = notificationDetails?.toMap();
+    await _channel.invokeMethod(
+        'show',
+        <String, Object>{
+          'id': id,
+          'title': title,
+          'body': body,
+          'platformSpecifics': serializedPlatformSpecifics,
+          'payload': payload ?? '',
+          if (matchDateTimeComponents != null)
+            'matchDateTimeComponents': matchDateTimeComponents.index
+        }
+          ..addAll(scheduledDate.toMap()));
+  }
+
+  @override
+  Future<void> cancel(int id) {
+    validateId(id);
+    if (_notificationButtonHandlerMap.remove(id) != null) {
+      return _channel.invokeMethod('cancel', id);
+    } else if (_periodicNotificationButtonHandlerMap.remove(id) != null) {
+      return _channel.invokeMethod('cancelPeriodicNotification', id);
+    } else {
+      throw ArgumentError.value(id, 'id', 'id is invalid');
+    }
+  }
+
+  @override
+  Future<void> cancelAll() async {
+    for (final int id in _notificationButtonHandlerMap.keys) {
```

Comment:
What happens if the user triggers a notification that is displayed, leaves the notification there and restarts the app? If I'm not mistaken, looks as though cancelling notifications and handling buttons would rely on having the same app session active. FYI that cancelling notifications and handling when the user taps on a notification (note: other platforms don't support action buttons at the moment) for all other platforms can be done even if the user restarted the app so if what I described is indeed how it works right now (as I'm still having some trouble with my Linux environment to test) then a solution will need to be found for Linux to behave the same. I imagine, this would potentially need to rely on persisting some information about the notification